nixos/modules/tasks/filesystems/nilfs2.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  inInitrd = any (fs: fs == "nilfs2") config.boot.initrd.supportedFilesystems;
+
+  # this file is mandatory for nilfs_cleanerd - can be empty
+  cleaner_conf = pkgs.writeText "nilfs_cleanerd.conf" ''
+    ${config.boot.nilfs2.cleanerd_conf}
+  '';
+
+in
+
+{
+  config = mkIf (any (fs: fs == "nilfs2") config.boot.supportedFilesystems) {
+
+    system.fsPackages = [ pkgs.nilfs-utils ];
+
+    boot.initrd.availableKernelModules = mkIf inInitrd [ "nilfs2" ];
+
+    boot.initrd.extraUtilsCommands = mkIf (inInitrd && !config.boot.initrd.systemd.enable)
+      ''
+        copy_bin_and_libs ${pkgs.nilfs-utils}/bin/chcp
+        copy_bin_and_libs ${pkgs.nilfs-utils}/bin/lscp
+        copy_bin_and_libs ${pkgs.nilfs-utils}/bin/mkcp
+        copy_bin_and_libs ${pkgs.nilfs-utils}/bin/mount.nilfs2
+        copy_bin_and_libs ${pkgs.nilfs-utils}/bin/nilfs-clean
+        copy_bin_and_libs ${pkgs.nilfs-utils}/bin/nilfs-resize
+        copy_bin_and_libs ${pkgs.nilfs-utils}/bin/rmcp
+      '';
+
+    environment.etc = {
+      "nilfs_cleanerd.conf".source = cleaner_conf;
+    };
+
+  };
+
+  options = {
+    boot.nilfs2 = {
+      cleanerd_conf = mkOption {
+        type = types.lines;
```

Comment:
Any interest in using an attribute set here instead?

Reply:
This would be better indeed

Reply:
it's using attributes now, way better :+1: